nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/model/DocumentModelResolver.java

```diff
@@ -177,16 +177,24 @@ public Object fetch(Object value) throws IllegalStateException {
                 try (CloseableCoreSession session = CoreInstance.openCoreSession(ref.repo)) {
                     try {
                         DocumentModel doc;
+                        DocumentRef documentRef;
+
                         switch (mode) {
                         case ID_REF:
-                            doc = session.getDocument(new IdRef(ref.ref));
+                            documentRef = new IdRef(ref.ref);
                             break;
                         case PATH_REF:
-                            doc = session.getDocument(new PathRef(ref.ref));
+                            documentRef = new PathRef(ref.ref);
                             break;
                         default:
                             throw new UnsupportedOperationException();
                         }
+
+                        if (!session.exists(documentRef)) {
+                            return null;
```

Comment:
Add a comment `// the document doesn't exist or is not accessible by the current user`